### PR TITLE
Phani | BAH-3743 | Ability to show display controls in consultation Medications tab

### PR DESCRIPTION
**Pull request by binduak**
JIRA Card -> [https://bahmni.atlassian.net/browse/BAH-3743](https://bahmni.atlassian.net/browse/BAH-3743)

### CODE REVIEW

File: ui/app/common/displaycontrols/dashboard/directives/dashboard.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 angular.module('bahmni.common.displaycontrol.dashboard')
-    .directive('dashboard', ['appService', '$stateParams', '$bahmniCookieStore', 'configurations', 'encounterService', 'spinner', 'auditLogService', 'messagingService', '$state', '$translate', function (appService, $stateParams, $bahmniCookieStore, configurations, encounterService, spinner, auditLogService, messagingService, $state, $translate) {
+    .directive('dashboard', ['appService', '$stateParams', '$bahmniCookieStore', 'configurations', 'encounterService', 'spinner', 'auditLogService', 'messagingService', '$state', '$translate', '$rootScope', function (appService, $stateParams, $bahmniCookieStore, configurations, encounterService, spinner, auditLogService, messagingService, $state, $translate, $rootScope) {
```
Comment (rahu1ramesh) [inline ui/app/common/displaycontrols/dashboard/directives/dashboard.js:4]: @kavitha-sundararajan I don't see `$rootScope` in use, anywhere in dashboard.js, can you validate if this is a required addition.

File: ui/app/clinical/consultation/controllers/treatmentController.js
Comment (rahu1ramesh) [inline ui/app/clinical/consultation/controllers/treatmentController.js:1]: @kavitha-sundararajan , it seems that `allMedicationTabConfig` is never initialized or used in the code. Could you please check if this is intentional or if a relevant commit/change has been missed?

File: ui/app/common/displaycontrols/dashboard/directives/dashboard.js (hunk 1 above)
Comment (binduak) [reply ui/app/common/displaycontrols/dashboard/directives/dashboard.js]: @rahu1ramesh handled the review comment

File: ui/app/clinical/consultation/controllers/treatmentController.js
Comment (binduak) [reply ui/app/clinical/consultation/controllers/treatmentController.js:1]: @rahu1ramesh will check and comeback on this one

File: ui/app/clinical/common/controllers/visitController.js
Comment (rahu1ramesh) [inline ui/app/clinical/common/controllers/visitController.js:1]: @binduak I noticed that below is a duplicate of what we have in https://github.com/Bahmni/openmrs-module-bahmniapps/blob/e2ce8c0cb4ad3773774821e83c7b4628495b3635/ui/app/clinical/consultation/controllers/drugOrderHistoryController.js#L141. To reduce the duplicate code, we can move the logic for fetching and processing allergies into the `allergyService`. This makes the code more modular and reusable. 

@binduak I don't foresee any issue arising from this minor refactoring, and I hope it's okay if I implement the same?

**Review by rahu1ramesh — APPROVED** [review]

**Pull request merged**